config/config.go

```diff
@@ -828,6 +836,47 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return nil
 }
 
+// GCESDConfig is the configuration for GCE based service discovery.
+type GCESDConfig struct {
+	// Project: The Google Cloud Project ID
+	Project string `yaml:"project"`
+
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple gce_sd_configs
+	Zone string `yaml:"zone"`
+
+	// Filter: Can be used optionally to filter the instance list by other criteria.
+	// Syntax of this filter string is described here in the filter query parameter section:
+	// https://cloud.google.com/compute/docs/reference/latest/instances/list
+	Filter string `yaml:"filter,omitempty"`
```

Comment:
If filtering can be enabled user-defined in such a straightforward way, I'm
voting for keeping it. Should still be exposed in labels to a sane degree.

On Tue, Jun 28, 2016, 8:17 PM Brian Brazil <notifications@github.com> wrote:

> In config/config.go
> <https://github.com/prometheus/prometheus/pull/1769#discussion_r68812638>:
>
> > @@ -828,6 +836,47 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
> >  	return nil
> >  }
> >
> > +// GCESDConfig is the configuration for GCE based service discovery.
> > +type GCESDConfig struct {
> > +	// Project: The Google Cloud Project ID
> > +	Project string `yaml:"project"`
> > +
> > +	// Zone: The zone of the scrape targets.
> > +	// If you need to configure multiple zones use multiple gce_sd_configs
> > +	Zone string `yaml:"zone"`
> > +
> > +	// Filter: Can be used optionally to filter the instance list by other criteria.
> > +	// Syntax of this filter string is described here in the filter query parameter section:
> > +	// https://cloud.google.com/compute/docs/reference/latest/instances/list
> > +	Filter string `yaml:"filter,omitempty"`
>
> You should expose all this information as metadata for relabelling. If
> it's useful for filtering, it's likely also useful as labels too.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/prometheus/pull/1769/files/aaca980fcffb0da8ac9d3711ab33e4abfb536610..aaf0f438d9011e3d02fff333c738551ab9d491d0#r68812638>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEuA8q16v37qvb_mNNiIs-uzq7HLVOVZks5qQWVQgaJpZM4JAGOO>
> .
>